src/crypto/tls/delegated_credentials.go

```diff
@@ -448,17 +464,29 @@ func (dc *DelegatedCredential) Validate(cert *x509.Certificate, isClient bool, n
 		}
 
 		return ed25519.Verify(pk, in, dc.signature)
+	case PSSWithSHA256,
+		PSSWithSHA384,
+		PSSWithSHA512:
+		pk, ok := cert.PublicKey.(*rsa.PublicKey)
+		if !ok {
+			return false
+		}
+		hash := getHash(dc.algorithm)
+		return rsa.VerifyPSS(pk, hash, in, dc.signature, nil) == nil
```

Comment:
Not particular to this PR, but it's a bit odd that we call the signed string `in` here but `value` above.

Reply:
I think the idea is that here `in` is an untrusted input we want to validate, but above it's a value we're preparing to sign / commit to. Not sure I particularly buy into the logic, but I think it does highlight that the function is being used in two different ways.